h/static/scripts/group-forms/components/CreateEditGroupForm.tsx

```diff
@@ -305,9 +304,18 @@ export default function CreateEditGroupForm({
             <Checkbox
               data-testid="pre-moderation"
               checked={preModerated}
-              onChange={e =>
-                setPreModerated((e.target as HTMLInputElement).checked)
-              }
+              onChange={e => {
+                const isNowChecked = (e.target as HTMLInputElement).checked;
+                setPreModerated(isNowChecked);
+
+                // Mark form as unsaved only if the initial value is different
+                // from the new one
+                setSaveState(
+                  isNowChecked !== initiallyPreModerated
+                    ? 'unsaved'
+                    : 'unmodified',
```

Comment:
This could reset the form state to "unmodified" even if there are other fields that are unsaved. If a user clicks a checkbox twice, I think it is fine to leave the form in an "unsaved" state, even though the old and new state are actually the same. In other words, it is sufficient for the save state to reflect whether there have been editing actions since the last save, rather than whether the old/new values are actually different.

A more elaborate approach would be to remember pending and last-saved values per form field and then compute the overall unsaved-state from that of all the individual fields. That's out of scope for this PR.

Reply:
ooohh, that's right. Good catch.

Reply:
I have followed the simple approach to avoid overcomplicating the logic here.